### PR TITLE
Fix Windows unix socket URL round-trip

### LINKED ISSUE #6291: Windows: AF_UNIX unix:// URLs fail to parse and round-trip
## Summary

On Windows, in-tree AF_UNIX discovery tests fail at HEAD (`3c4dec3`). Named-pipe (`npipe://`) tests on the same machine pass. No container runtime is involved.

I intend to implement a fix. Per CONTRIBUTING I will wait for assignment before opening a PR.

## Environment

- Windows 11, `go version go1.26.5 windows/amd64`
- HEAD: `3c4dec3`

```
go test ./pkg/server/discovery/ -count=1 -run "TestParseUnixSocketPath_Valid$|TestHTTPClientForURL_SchemeDispatchCaseInsensitive|TestCheckHealth_UnixSocket_Success$"
```

### POSIX `unix:///` paths rejected as relative

`ParseUnixSocketPath("unix:///var/run/thv.sock")` returns:

```
unix socket path must be absolute: \var\run\thv.sock
```

`filepath.Clean` on Windows turns `/var/run/thv.sock` into `\var\run\thv.sock`, and `filepath.IsAbs` is false because there is no volume. Same for `HTTPClientForURL("UNIX:///tmp/thv.sock")` -> `\tmp\thv.sock`.

### Concatenating `unix://` + a drive path is not a valid URL

`TestCheckHealth_UnixSocket_Success` builds `"unix://"+socketPath`. On Windows that is `unix://C:\Users\...\test.sock`. `url.Parse` rejects it:

```
invalid server URL: parse "unix://C:\\Users\\stans\\AppData\\Local\\Temp\\thv-3078206842\\test.sock": invalid port ":\\Users\\stans\\AppData\\Local\\Temp\\thv-3078206842\\test.sock" after host
```

`pkg/api/socket_windows.go` already documents this and emits `unix:///C:%5Cpath%5Cthv.sock`. The discovery tests still concatenate.

### `ListenURL` four slashes

Windows `socketURL` does `Path: "/" + address`. When `address` is already `/tmp/test.sock`, `ListenURL` returns `unix:////tmp/test.sock` (four slashes) instead of `unix:///tmp/test.sock`. The Unix build of `socketURL` uses `Path: address` and does not add that extra slash.

## What is not broken

- Named-pipe setup, DACL, and first-instance-wins tests are green.
- `TestSocketURL_RoundTrip_AFUnix` for `C:\path\thv.sock` via `socketURL()` already percent-encodes correctly.

## Direction (after assignment)

Keep the three-slash encoded form that `socket_windows.go` already documents. Make `ParseUnixSocketPath` accept POSIX `unix:///var/run/...` on Windows (or skip those cases on Windows). Stop tests from concatenating `unix://` + a drive path; use `socketURL` / `url.URL`. Do not prepend `/` when `address` already starts with `/`.

Happy to take this once assigned.

Filed with AI assistance (Cursor/Grok). Reproduced on this machine.

### PR DESCRIPTION
**Pull request by stantheman0128**
## Summary

I reproduced this on Windows 11 with `go test ./pkg/api ./pkg/server/discovery`. Drive-letter emit was already `unix:///C:%5C...`. What still failed:

- `ListenURL` for `/tmp/test.sock` came out as `unix:////tmp/test.sock` (extra slash).
- `ParseUnixSocketPath("unix:///var/run/thv.sock")` ran the path through `filepath.Clean`, so Windows turned it into `\var\run\thv.sock` and rejected it as not absolute.
- Health tests built `"unix://"+socketPath`, which `url.Parse` rejects on a drive letter (`invalid port`).

I put emit on one helper (`discovery.UnixSocketURL`) and taught the parser the two absolute forms instead of three one-off patches.

- POSIX paths keep three slashes. Drive-letter paths still get the synthetic leading slash that `net/url` needs.
- POSIX `unix:///` stays slash-form on Windows. The old four-slash alias still parses.
- Health tests call the helper.

Fixes #6291

## Type of change

- [x] Bug fix
- [ ] New feature
- [ ] Refactoring (no behavior change)
- [ ] Dependency update
- [ ] Documentation
- [ ] Other (describe):

## Test plan

- [ ] Unit tests (`task test`)
- [ ] E2E tests (`task test-e2e`)
- [ ] Linting (`task lint-fix`)
- [x] Manual testing (describe below)

Win11, go1.26.5, worktree at `origin/main` `e25de16`. I did not run full `task test` (whole-repo `-race`). Focused:

```
go test -timeout 90s -count=1 ./pkg/server/discovery/
ok  	github.com/stacklok/toolhive/pkg/server/discovery	1.511s

go test -timeout 90s -count=1 ./pkg/api/ -run "TestListenURL|TestSocketURL"
ok  	github.com/stacklok/toolhive/pkg/api	4.847s

go test -timeout 60s -count=1 ./pkg/api/ -run "TestSetupUnixSocket|TestCleanupUnixSocket|TestCreateListener|TestIsNamedPipe"
ok  	github.com/stacklok/toolhive/pkg/api	1.817s
```

`TestListenURL/Unix_socket_returns_unix_URL`, `TestUnixSocketURL`, POSIX/drive-letter round-trip, and `TestParseUnixSocketPath_FourSlashAlias` are in that set.

## API Compatibility

- [x] This PR does not break the `v1beta1` API, OR the `api-break-allowed` label is applied and the migration guidance is described above.

## Changes

| File | Change |
|------|--------|
| `pkg/server/discovery/unix_url.go` | Shared `UnixSocketURL` plus POSIX vs drive-letter parse |
| `pkg/server/discovery/health.go` | `ParseUnixSocketPath` delegates to that parse |
| `pkg/api/socket_windows.go` / `socket_unix.go` | `socketURL` calls `UnixSocketURL` (npipe branch unchanged) |
| `*_test.go` | Round-trip, four-slash alias, health tests stop concatenating |

## Does this introduce a user-facing change?

Yes, for AF_UNIX discovery URLs. A POSIX socket path on Windows is now `unix:///tmp/...` (three slashes), not `unix:////tmp/...`. Drive-letter URLs are unchanged (`unix:///C:%5C...`). Named pipes are unchanged.

## Special notes for reviewers

I left `pkg/container/docker/sdk/client_unix.go` (`unix://`+docker socket into moby `WithHost`) alone. That is a different host string, unix-only build.

I used Cursor to write the patch after reproducing the failures on this box.

Claimed and assigned on #6291.

### CODE REVIEW

File: pkg/server/discovery/unix_url.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+func parseDriveLetterSocketPath(p string) (string, error) {
+	if p[0] == '/' {
+		p = p[1:]
+	}
+	if strings.Contains(p, "..") {
+		return "", fmt.Errorf("unix socket path must not contain '..': %s", p)
+	}
+	cleaned := filepath.Clean(p)
+	if !isDriveLetterPath(cleaned) {
```
Comment (samuv) [inline pkg/server/discovery/unix_url.go]: Could we keep rejecting drive-relative paths here? `isDriveLetterPath` only checks for `<letter>:`, so `ParseUnixSocketPath("unix:///C:relative.sock")` now succeeds and returns `C:relative.sock`. On Windows this is relative to the process's current directory for drive C, which can make the health client dial a different socket even though this API promises an absolute path. The previous `filepath.IsAbs` check rejected this form. Requiring `/` or `\` after the colon, with regression cases for `unix:///C:relative.sock` and `unix:///C:`, should preserve the absolute-path contract.

**Review by samuv — COMMENTED** [review]
Thanks for the focused fix and the coverage for both POSIX and drive-letter round trips. I found one path-validation edge case where a drive-relative Windows path is accepted as absolute, so I left an inline comment. The current PR checks are green.

Checklist:
- Tests: New regression tests cover the intended paths, and CI is green.
- Docs: No additional docs needed for this internal compatibility fix.
- Registry impact: None.
- Security: One path-validation issue noted inline.
- Backwards compatibility: The legacy four-slash alias remains supported.

File: pkg/server/discovery/unix_url.go (hunk 1 above)
Comment (stantheman0128) [reply pkg/server/discovery/unix_url.go]: Drive-relative C:foo is not absolute. Parse now requires a slash or backslash after the colon. unix:///C:relative.sock and unix:///C: are in the tests.

Comment (samuv) [reply pkg/server/discovery/unix_url.go]: nice!

**Review by samuv — APPROVED** [review]
